### ISSUE TITLE
.send becomes .query for GET requests, but will error out when used for HEAD requests

### ISSUE THREAD

**Issue by mikemaccana** [visionmedia/superagent#534]
From the docs:

> When issuing a GET request the res.send(obj) method will invoke res.query(obj),

For consistency's sake, the same should apply to HEAD requests, but doesn't. When using `.send` rather than `.query` on a HEAD request superagent will error:

```
http.js:851
    throw new TypeError('first argument must be a string or Buffer');
          ^
TypeError: first argument must be a string or Buffer
    at ClientRequest.OutgoingMessage.write (http.js:851:11)
    at ClientRequest.OutgoingMessage.end (http.js:981:16)
    at Request.end (/Users/someuser/Documents/someproject/node_modules/superagent/lib/node/index.js:968:9)
    at Object.isDuplicate (/Users/someuser/Documents/someproject/node_modules/mycognition-usermanagement/index.js:44:5)
    at Object.<anonymous> (/Users/someuser/Documents/someproject/node_modules/mycognition-usermanagement/example.js:3:27)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
```

**Comment by englercj** [comment]
I also don't think this is true for the browser version, which seems to operate differently. I find I have to actually call `query` instead of `send` for the browser version.

**Comment by acontreras89** [comment]
What's the status on this? 

Perhaps [this line of code](https://github.com/visionmedia/superagent/blob/master/lib/client.js#L810) should be changed to:

```js
if (data) req.query(data); // rather than req.send
```

As described by [the HTTP spec](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.4) (credits to [this SO answer](http://stackoverflow.com/questions/4529066/post-data-with-head-request#answer-4529097)):

> The HEAD method is identical to GET except that the server MUST NOT return a message-body in the response.

It would be nice to have both functions (`.get` and `.head`) behave the same way.

**Comment by focusaurus** [comment]
God I wish our APIs would just consistently do the same thing instead of overloading generic methods like `send` and trying to be clever. That said, if we behave that way and document it for GET, I'd agree to at least make it consistent for HEAD, although ultimately I'd rather not have a method that does 2 totally different things depending on the state of the request built up in the method chaining DSL so far.

**Comment by focusaurus** [comment]
OK so we're all on the same page, here's the current behavior and some options for changes we could make

- Browser `.get(url, payload, cb)`
  - Currently puts payload in the query string
- Browser `.head(url, payload, cb)`
  - Currently puts payload in the body
  - Proposed change to put payload in the query string
- Node `.get(url, payload, cb)`
  - Currently puts payload in the body
  - Proposed change to put payload in the query string
- Browser `.head(url, payload, cb)`
  - Currently puts payload in the body
  - Proposed change to put payload in the query string

I think whatever we choose, if technically possible, we should be consistent across the full matrix above. I'm OK with either of the below choices.

**Option A**: make them all use the query string. I think this would match what I believe was probably TJ's intent when he coded the `.get()` implementation in f3955101 back in 2012. We'd need another commit on the node side and probably some test updates to get here.

**Option B**: make them all use the body. I believe technically GET and HEAD requests are permitted to have bodies although they typically do not. This would be a breaking semantic change for 1/4 of the API surface above, and a blessing of the existing behavior as the intended for the other 3/4.

**Comment by pornel** [comment]
I'm OK with switch to query for the shorthand function (I'm assuming explicit `.get().send()` would continue to work for those who really want a body).

**Comment by kornelski** [comment]
I think that's fixed in #1228
